### ISSUE TITLE
How possible can be to read a collection without an Azure Function?

### ISSUE THREAD

**Issue by ylr-research** [JeremyLikness/jeremylikness-blog#10]
Hello @JeremyLikness ,

I have a Blazor WASM (POC) that needs to retrieve a list of books on sale from Azure CosmosDB. This list is public, thus no user needs to be authenticated to see it. Everything was fine up to .NET 5 came along and I upgraded the project. Azure CosmosDB SDK started to throw `System.PlatformNotSupportedException` (see [here](https://github.com/Azure/azure-cosmos-dotnet-v3/issues/1865)). After opening some issues Dan [recommended](https://github.com/dotnet/aspnetcore/issues/26942) to read your [blog](content/blog/azure-ad-secured-serverless-cosmosdb-from-blazor-webassembly/index.md) which he considers the preferred approach. However, if I'm not mistaken, your scenario assumes a user will be authenticated, which is not my case. 

Basically, I'm wondering in my scenario whether is possible or not to avoid using the Azure Function.

**Comment by JeremyLikness** [comment]
You still need a context to access Azure Cosmos DB; it has to "know" what permissions are taking place. In this case, you would still want a function to generate a token, but you don't have to secure it with AAD. You could reuse a permission set or generate a unique one with read permissions and use that.
